cdap-app-fabric/src/main/java/co/cask/cdap/config/PreferencesWrapper.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.config;
+
+import co.cask.cdap.api.common.Bytes;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+
+import java.util.Map;
+
+/**
+ * Wrapper around {@link ConfigStore} specifically for Preferences API.
+ */
+public class PreferencesWrapper {
+  private static final String PREFERENCES = "preferences";
+  private static final String INSTANCE_NAME = "global";
+  private static final String NAMESPACE = "default";
```

Comment:
Hmmm, I think we should discuss this more with @anew and @sreevatsanraman. The default namespace is a specific namespace. 

For what you want here (i.e. instance-level), I'd imagine you mean something that cuts across all namespaces. However, the default namespace is no different from any other namespace in the sense that it does not represent things that are common for all namespaces, its a namespace in itself.

We have come across a couple of things that may be instance-wide yet - like the /status or /instances endpoints or system datasets. I do not think we've settled on a strategy to handle them yet.